server/src/main/java/org/opensearch/search/profile/query/ConcurrentQueryProfileBreakdown.java

```diff
@@ -47,11 +51,62 @@ public Map<String, Long> toBreakdownMap() {
         final Map<String, Long> map = new HashMap<>(super.toBreakdownMap());
 
         for (final AbstractProfileBreakdown<QueryTimingType> context : contexts.values()) {
-            for (final Map.Entry<String, Long> entry : context.toBreakdownMap().entrySet()) {
-                map.merge(entry.getKey(), entry.getValue(), Long::sum);
+            Map<String, Long> breakdown = context.toBreakdownMap();
+            for (final Map.Entry<String, Long> entry : breakdown.entrySet()) {
+                if (entry.getKey().contains(TIMING_TYPE_COUNT_SUFFIX)) {
+                    map.merge(entry.getKey(), entry.getValue(), Long::sum);
+                }
             }
+            addMaxEndTimeAndMinStartTime(map, breakdown);
         }
+        return buildQueryProfileBreakdownMap(map);
+    }
 
-        return map;
+    /**
+     * This method is used to bring the max end time and min start time fields to the query profile breakdown.
+     *
+     * @param  map  the query level timing count breakdown for current instance
+     * @param  breakdown the segment level timing count breakdown for current instance
+     */
+    static void addMaxEndTimeAndMinStartTime(Map<String, Long> map, Map<String, Long> breakdown) {
+        for (QueryTimingType queryTimingType : QueryTimingType.values()) {
+            String timingType = queryTimingType.toString();
+            String timingTypeStartTime = timingType + AbstractProfileBreakdown.TIMING_TYPE_START_TIME_SUFFIX;
+            long currentMaxEndTime = map.getOrDefault(timingType + MAX_END_TIME_SUFFIX, 0L);
+            long currentMinStartTime = map.getOrDefault(timingType + MIN_START_TIME_SUFFIX, Long.MAX_VALUE);
+            // Only "create_weight" is performed at the query level
+            Map<String, Long> source = timingType.equals(QueryTimingType.CREATE_WEIGHT.toString()) ? map : breakdown;
+            long maxEndTime = Math.max(currentMaxEndTime, source.get(timingTypeStartTime) + source.get(timingType));
+            long minStartTime = Math.min(currentMinStartTime, source.get(timingTypeStartTime));
+            map.put(timingType + MAX_END_TIME_SUFFIX, maxEndTime);
+            map.put(timingType + MIN_START_TIME_SUFFIX, minStartTime);
```

Comment:
@ticheng-aws This mechanism will just tell us across slices what is the total time consumed for each operation including any wait time for slice execution. We will loose information like the time taken to just execute each operation in a slice (like min/max/avg scoring time or next doc time, etc). For this I think we can do following, as we will need a mechanism to create association between `leafReaderContext` and slice it is part of 

1. In `ProfileWeight`  we can have a method something like 'associateCollectorToLeaves(collector, leaves)`. This will internally call same method on the `ContextualProfileBreakdown`. The concurrent implementation of ProfileBreakdown will implement that method to keep track of collector ---> list of leaves (which is slice to leaves mapping). For non concurrent case it can be no-op.
2. Then during the `breakdownMap` computation we can use this info to compute the slice level timers or just have the start timer for each timing type in `breakdownMap` similar to AggregationProfiler.
3. When `doGetTree()` is called on QueryProfiler to fetch the result, the concurrent implementation of `QueryProfiler` can override (again similar to `ConcurrentAggregationProfiler`) and compute all the slice level timers and return the tree.

Reply:
Hey @sohami, I have now added slice-level statistics. Kindly assist in reviewing it once more, thank you.